src/handlers/track_table.rs

```diff
@@ -149,6 +152,108 @@ pub fn handler(key: Key, app: &mut App) {
   }
 }
 
+fn play_random_song(app: &mut App) {
+  let TrackTable { context, .. } = &app.track_table;
+  match &context {
+    Some(context) => match context {
+      TrackTableContext::MyPlaylists => {
+        let (context_uri, track_json) = match (&app.selected_playlist_index, &app.playlists) {
+          (Some(selected_playlist_index), Some(playlists)) => {
+            if let Some(selected_playlist) = playlists.items.get(selected_playlist_index.to_owned())
+            {
+              (
+                Some(selected_playlist.uri.to_owned()),
+                selected_playlist.tracks.get("total"),
+              )
+            } else {
+              (None, None)
+            }
+          }
+          _ => (None, None),
+        };
+
+        if let Some(val) = track_json {
+          let num_tracks: usize = from_value(val.clone()).unwrap();
+          app.dispatch(IoEvent::StartPlayback(
+            context_uri,
+            None,
+            Some(thread_rng().gen_range(0, num_tracks)),
+          ));
+        }
+      }
```

Comment:
Looks like this can be simplified 
```suggestion
        if let (Some(selected_playlist_index), Some(playlists)) =
          (app.selected_playlist_index, &app.playlists)
        {
          if let Some(selected_playlist) = playlists.items.get(selected_playlist_index) {
            let num_tracks = playlists.total as usize;
            let context_uri = selected_playlist.uri.clone();
            app.dispatch(IoEvent::StartPlayback(
              Some(context_uri),
              None,
              Some(thread_rng().gen_range(0, num_tracks)),
            ));
          };
        };

```

Reply:
playlist.tracks is a HashMap of sedre_json::Value.
rspotify doesn't fully unpack the response from the server, the real fix would be add the functionality to rspotify.